app/controllers/admin/cases_controller.rb

```diff
@@ -30,55 +34,118 @@ def index
   end
 
   def new
-    case_creator = CTS::Cases::Create.new(Rails.logger, case_model: Case::Base)
+    if params[:correspondence_type].present?
+      @correspondence_type = params[:correspondence_type]
+      self.send("prepare_new_#{@correspondence_type}".to_sym)
+    else
+      prepare_select_type
+    end
+  end
+
+  def prepare_new_foi
+    @correspondence_type_abbreviation = params[:correspondence_type]
+    case_class = correspondence_types_map[@correspondence_type_abbreviation.to_sym].first
+    @case = case_class.new
+
+    case_creator = CTS::Cases::Create.new(Rails.logger, case_model: Case::Base, type: 'Case::FOI::Standard' )
     @case = case_creator.new_case
-    @case.responding_team = BusinessUnit.responding.sample
+    @case.responding_team = BusinessUnit.responding.responding_for_correspondence_type(CorrespondenceType.foi).sample
     @case.flag_for_disclosure_specialists = 'no'
     @target_states = available_target_states
     @selected_state = 'drafting'
     @s3_direct_post = S3Uploader.s3_direct_post_for_case(@case, 'requests')
+
+    render :new
+  end
+
+  def prepare_new_sar
+    @correspondence_type_abbreviation = params[:correspondence_type]
+    case_class = correspondence_types_map[@correspondence_type_abbreviation.to_sym].first
+    @case = case_class.new
+
+    case_creator = CTS::Cases::Create.new(Rails.logger, case_model: Case::Base, type: 'Case::SAR' )
+    @case = case_creator.new_case
+    @case.responding_team = BusinessUnit.responding.responding_for_correspondence_type(CorrespondenceType.sar).sample
+    @target_states = available_target_states
+    @selected_state = 'drafting'
+    @s3_direct_post = S3Uploader.s3_direct_post_for_case(@case, 'requests')
+    @case.reply_method = 'send_by_email'
+
     render :new
   end
 
   private
 
+  def prepare_select_type
```

Comment:
What is being prepared here?

Reply:
Nothing this pattern was copied from cases_controller (the non-admin one) but happy to change make more sense!